cogs/welcome/welcome.py

```diff
@@ -537,6 +644,104 @@ def check(message: discord.Message):
+
+        """
+        await self.ensureCurrentServerHasImgCache(ctx.channel)
+        file_name = f"{name}.png"
+        fp = os.path.join(self.imgDir, str(ctx.channel.guild.id), file_name)
+
+        if os.path.exists(fp):
+            await ctx.reply(
+                "This name is already in use! For ease of management, please use another name."
+            )
+            return
+
+        image = None
+        if len(ctx.message.attachments) == 1:
+            image = ctx.message.attachments[0]
+
+        else:
+            await ctx.reply(
+                "You need to attach exactly 1 image in the message that uses this command"
+            )
+            return
+
+        await image.save(fp)
+
+        # Performing necessary checks to ensure that this base can produce a good generated image
+        temp = Image.open(fp)
+        temp_resize = temp.resize((1193, 671), 2)
+        temp_resize.save(fp, dpi=(72, 72))
+
+        # alert user that their image has been added
+        await ctx.reply("Image added to this server's image cache")
+
+    # [p]welcomeset greetings image remove
+    @image.command(name="remove")
+    async def imgRemove(self, ctx: Context, img_name: str):
+        """Removes the specified image from the pool"""
+        await self.ensureCurrentServerHasImgCache(ctx.channel)
+        file_name = f"{img_name}.png"
+        try:
+            os.remove(os.path.join(self.imgDir, str(ctx.channel.guild.id), file_name))
+        except:
+            await ctx.reply("The named image doesn't exist")
+            return
+
+        if len(os.listdir(os.path.join(self.imgDir, str(ctx.channel.guild.id)))) == 0:
+            self.config.guild(ctx.guild).get_attr(KEY_TOGGLE_RANDOM_MSG).set(False)
+            await ctx.reply("Last image deleted. Image randomiser turned off.")
+
+        await ctx.reply("Image sucessfully removed")
+
+    # [p]welcomeset greetings image view
+    @image.command(name="view")
+    async def showImg(self, ctx: Context, img_name: str):
+        """Show the named image from the image pool if it exists"""
+        await self.ensureCurrentServerHasImgCache(ctx.channel)
+        file_name = f"{img_name}.png"
+        try:
+            await ctx.send(
+                img_name + ":",
+                file=discord.File(os.path.join(self.imgDir, str(ctx.channel.guild.id), file_name)),
+            )
+        except:
+            await ctx.reply("The named image doesn't exist")
+
+    # [p]welcomeset greetings image list
+    @image.command(name="list")
+    async def listImg(self, ctx: Context):
+        """Display a list of all the images in this server's image cache"""
+        await self.ensureCurrentServerHasImgCache(ctx.channel)
+        listOfImages = "\n".join(os.listdir(os.path.join(self.imgDir, str(ctx.channel.guild.id))))
+        if len(listOfImages) == 0:
+            await ctx.reply("No images added yet.")
+            return
+        await ctx.reply(listOfImages.replace(".png", ""))
```

Comment:
Here we can use `pathlib.Path` in this fashion:

```python
listOfImages = "\n".join(imagePath.stem for imagePath in pathlib.Path("some path").iterdir())
```

This way we won't need to manually remove file extensions.